Spec/Auth.swift

```diff
@@ -1999,6 +1999,154 @@ class Auth : QuickSpec {
                     }
                 }
             }
+            
+            // RSA10k
+            context("should adhere to all requirements relating to") {
+                it("Should obtain server time once and persist the offset from the local clock") {
+                    let rest = ARTRest(options: AblyTests.commonAppSetup())
+                    
+                    var serverTimeRequestWasMade = false
+                    let block: @convention(block) (AspectInfo) -> Void = { _ in
+                        serverTimeRequestWasMade = true
+                    }
+                    
+                    let hook = ARTRest.aspect_hookSelector(rest)
+                    // Adds a block of code after `time` is triggered
+                    let _ = try? hook(#selector(ARTRest.time(_:)), withOptions: .PositionAfter, usingBlock:  unsafeBitCast(block, ARTRest.self))
+                    
+                    let authOptions = ARTAuthOptions()
+                    authOptions.queryTime = true
+                    
+                    waitUntil(timeout: testTimeout) { done in
+                        rest.auth.createTokenRequest(nil, options: authOptions, callback: { tokenRequest, error in
+                            expect(error).to(beNil())
+                            guard let tokenRequest = tokenRequest else {
+                                XCTFail("TokenRequest is nil"); return
+                            }
+                            expect(tokenRequest.timestamp).toNot(beNil())
+                            expect(serverTimeRequestWasMade).to(beTrue())
+                            expect(rest.auth.timeOffset).toNot(beNil())
+                            done()
+                        })
+                    }
+                    
+                    var serverTimeRequestWasMade2 = false
+                    let block2: @convention(block) (AspectInfo) -> Void = { _ in
+                        serverTimeRequestWasMade2 = true
+                    }
+                    
+                    let hook2 = ARTRest.aspect_hookSelector(rest)
+                    let _ = try? hook2(#selector(ARTRest.time(_:)), withOptions: .PositionAfter, usingBlock:  unsafeBitCast(block2, ARTRest.self))
+                    
+                    waitUntil(timeout: testTimeout) { done in
+                        rest.auth.createTokenRequest(nil, options: authOptions, callback: { tokenRequest, error in
+                            expect(error).to(beNil())
+                            guard let tokenRequest = tokenRequest else {
+                                XCTFail("TokenRequest is nil"); return
+                            }
+                            expect(tokenRequest.timestamp).toNot(beNil())
```

Comment:
You should also check that this has the expected value within a margin of error, ie. the first token request's timestamp plus the some offset. (You can wrap this in a `delay(1) { ... }` to make the difference more apparent.)